lib/project_types/extension/commands/serve.rb

```diff
@@ -39,6 +40,13 @@ def argo_admin?
           ShopifyCli::Feature.enabled?(:argo_admin_beta) &&
           extension_type.specification.features&.argo&.surface == "admin"
       end
+
+      def validate_env
+        ExtensionProject.reload
```

Comment:
I need to do reload here just because there are many places writing env file to disk, not just `extension_project.rb`.

The `ensure_env.rb` write env to disk but didn't (cannot) reload ExtensionProject https://github.com/shopify/shopify-app-cli/blob/master/lib/shopify-cli/tasks/ensure_env.rb#L84.